microsoft/testsuites/security/openssl.py

```diff
@@ -66,3 +81,64 @@ def _openssl_test_encrypt_decrypt(self, log: Logger, node: Node) -> None:
             assert_that(plaintext).described_as(
                 "Plaintext and decrypted data do not match"
             ).is_equal_to(decrypted_data)
+
+    def _openssl_test_sign_verify(self, log: Logger, node: Node) -> None:
+        """
+        Tests OpenSSL signing and verification functionality.
+        This function generates a key pair, signs a message,
+        and verifies the signature.
+        """
+        openssl = node.tools[OpenSSL]
+        private_key, public_key = openssl.create_key_pair()
+
+        plaintext = "cool"
+        signature = openssl.sign(plaintext, private_key)
+        openssl.verify(plaintext, public_key, signature)
+
+        log.debug("Successfully signed and verified a file.")
+
+    @TestCaseMetadata(
+        description="""
+        This test will use Go experimental system crypto tests
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[CBLMariner],
+        ),
+    )
+    def openssl_golang__sys_crypto_tests_verify(self, log: Logger, node: Node) -> None:
+        """Verifies the Go experimental system crypto tests"""
+        self._run_go_crypto_tests(log, node)
+        return
+
+    def _run_go_crypto_tests(self, log: Logger, node: Node) -> None:
+        """
+        This test sets up the dependencies to run the
+        experimental Go system crypto tests and cleans go builds.
+        """
+        # installs go dependencies for tests
+        az_os = cast(CBLMariner, node.os)
```

Comment:
Why does it need to convert?

Reply:
I see, casting to POSIX is better than CBLMariner. The test case may extend to other distros in the future.